contracts/oracle/definer-oracle/ExOracleTPT.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity 0.5.14;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../IExOraclePriceData.sol";
+
+/**
+ * @title TPT Oracle
+ * @dev Contract represent TPT ExOracle pair.
+ */
+contract ExOracleTPT is Ownable {
+    using SafeMath for uint256;
+
+    // ExOracle contract address
+    IExOraclePriceData public exOracleAddress;
+
+    // ExOracle DataSource
+    address public dataSource;
+
+    // TPT token price in USD, with 6 decimal places
+    uint256 public tptPriceInUSD;
+
+    // TPT token last updated timestamp
+    uint256 public tptLastUpdateTimestamp;
+
+    // Pair Name: example: "DAI/ETH"
+    string public constant pairName = "TPT/OKT";
+
+    // Price Type required by ExOracle. Example: "DAI"
+    string public constant priceType = "TPT";
+
+    uint256 public constant ONE_OKT = 10 ** 18;
+    // 6 decimals USD rate returned by ExOracle, multiplier, divisor
+    uint256 public constant USD_DECIMALS_MUL_DIV = 10 ** 6;
+    uint256 public constant OKT_NUMERATOR = ONE_OKT * USD_DECIMALS_MUL_DIV;
+
+    event PriceUpdated(uint256 priceInUSD, uint256 timestamp);
```

Comment:
I think this event should be called `TPTPriceUpdated`, so we don't confuse while adding more tokens in the future

Reply:
fixed